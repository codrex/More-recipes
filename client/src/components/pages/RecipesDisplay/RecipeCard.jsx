import React from 'react';
import PropTypes from 'prop-types';
import { Image } from 'cloudinary-react';
import classnames from 'classnames';
import Icon from '../../common/Icon';
import Button from '../../common/Button';
import Animation from '../../common/Animation';
import { DEFAULT_RECIPE_PIX } from '../../../constants';


const RecipeCard = (props) => {
  const fav = props.isFav(props.recipeId);
  const {
    recipeName,
    category,
    recipeId,
    downVotes,
    upVotes,
    views,
    push,
    image
  } = props;
  return (
    <Animation>
      <div className="recipe-card">
        <div className="recipe-card-body">
          <Image
            cloudName="resycom"
            publicId={image.publicId || DEFAULT_RECIPE_PIX}
            width="400"
            height="250"
            crop="fill"
            className="recipe-card-img"
          />
          <div className="recipe-card-underlay">
            <p className="lead text-capitalize">{category}</p>
            <Button
              handleClick={() => {
                push(`/recipe/${recipeId}`);
              }}
              text="view details"
              className="recipe-card-underlay-btn"
            />
          </div>
        </div>
        <div className="recipe-card-footer">
          <h4 className="ellipsis">{recipeName}</h4>
          <div className="d-flex col-12 no-padding justify-content-between">
            <div className="d-flex col-8 no-padding">
              <Icon iconClass="fa fa-thumbs-up recipe-card-icon ">
                {upVotes}
              </Icon>
              <Icon iconClass="fa fa-thumbs-down recipe-card-icon ">
                {downVotes}
              </Icon>
              <Icon iconClass="fa fa-eye recipe-card-icon">{views}</Icon>
            </div>
            <Icon
              iconClass={classnames('fa recipe-card-icon fav-color fav',
                !fav && 'fa-heart-o', fav && 'fa-heart')}
              pointer
              handleClick={() => props.toggleFav(recipeId)}
            />
          </div>
        </div>
      </div>
    </Animation>
  );
};

RecipeCard.propTypes = {
  upVotes: PropTypes.number.isRequired,
  downVotes: PropTypes.number.isRequired,
  views: PropTypes.number.isRequired,
  recipeName: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  recipeId: PropTypes.number.isRequired,
  toggleFav: PropTypes.func.isRequired,
  isFav: PropTypes.func.isRequired,
  push: PropTypes.func.isRequired,
  image: PropTypes.oneOfType([PropTypes.object, PropTypes.string])
};

RecipeCard.defaultProps = {
  image: {}
};

export default RecipeCard;
