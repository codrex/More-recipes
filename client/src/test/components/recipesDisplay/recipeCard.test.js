/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import toJson from 'enzyme-to-json';
import RecipeCard from '../../../components/pages/RecipesDisplay/RecipeCard';

const props = {
  recipeName: 'rice and beans',
  category: 'dinner',
  views: 120,
  upVotes: 30,
  downVotes: 0,
  recipeId: 1,
  toggleFav: jest.fn(),
  isFav: jest.fn(),
  push: jest.fn(),
};

describe('Recipes card component ', () => {
  let component;
  beforeEach(() => {
    component = mount(<RecipeCard {...props} />);
  });

  test('expected to match snapshot', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  test('expected to match snapshot when favorite icon is clicked ', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
    component.find('.fav').simulate('click');
    expect(tree).toMatchSnapshot();
    expect(props.toggleFav).toBeCalled();
  });

  test('expected to match snapshot when view recipe button is clicked ',
    () => {
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
      component.find('Button.recipe-card-underlay-btn').simulate('click');
      expect(tree).toMatchSnapshot();
      expect(props.toggleFav).toBeCalled();
    });
});
