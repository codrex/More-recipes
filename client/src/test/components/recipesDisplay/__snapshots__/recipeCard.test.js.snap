// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes card component  render as expected 1`] = `
<span>
  <div
    className="recipe-card"
  >
    <div
      className="recipe-card-body"
    >
      <img
        className="recipe-card-img"
        height="250"
        src="http://res.cloudinary.com/resycom/image/upload/c_fill,h_250,w_400/recipe-card-placeholder_ta9ikp_iebsf2.jpg"
        width="400"
      />
      <div
        className="recipe-card-underlay"
      >
        <p
          className="lead text-capitalize"
        >
          dinner
        </p>
        <button
          className="btn recipe-card-underlay-btn"
          data-dismiss="modal"
          data-target=""
          data-toggle=""
          href=""
          id=""
          onClick={[Function]}
          type="button"
        >
          view details
        </button>
      </div>
    </div>
    <div
      className="recipe-card-footer"
    >
      <h4
        className="ellipsis"
      >
        rice and beans
      </h4>
      <div
        className="d-flex col-12 no-padding justify-content-between"
      >
        <div
          className="d-flex col-8 no-padding"
        >
          <div
            className="icon-wrapper"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="button"
            tabIndex="0"
          >
            <i
              className="icon fa fa-thumbs-up recipe-card-icon "
              data-for="1upvote"
              data-target=""
              data-tip="Total Upvotes: 30"
              data-toggle=""
            >
              <span>
                30
              </span>
            </i>
            <div
              className="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
            >
              
            </div>
          </div>
          <div
            className="icon-wrapper"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="button"
            tabIndex="0"
          >
            <i
              className="icon fa fa-thumbs-down recipe-card-icon "
              data-for="1downvote"
              data-target=""
              data-tip="Total Downvotes: 0"
              data-toggle=""
            >
              <span>
                0
              </span>
            </i>
            <div
              className="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
            >
              
            </div>
          </div>
          <div
            className="icon-wrapper"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="button"
            tabIndex="0"
          >
            <i
              className="icon fa fa-eye recipe-card-icon"
              data-for="1views"
              data-target=""
              data-tip="Total Number of Views: 120"
              data-toggle=""
            >
              <span>
                120
              </span>
            </i>
            <div
              className="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
            >
              
            </div>
          </div>
        </div>
        <div
          className="icon-wrapper pointer"
          onClick={[Function]}
          onMouseEnter={[Function]}
          role="button"
          tabIndex="0"
        >
          <i
            className="icon fa recipe-card-icon fav-color fav fa-heart-o"
            data-for="1fav"
            data-target=""
            data-tip="Favourite This Recipe"
            data-toggle=""
          >
            <span />
          </i>
          <div
            className="__react_component_tooltip place-top type-dark "
            data-id="tooltip"
          >
            
          </div>
        </div>
      </div>
    </div>
  </div>
</span>
`;

exports[`Recipes card component  render as expected when favorite icon is clicked  1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <RecipeCard
      category="dinner"
      downVotes={0}
      image={Object {}}
      isFav={[Function]}
      push={[Function]}
      recipeId={1}
      recipeName="rice and beans"
      toggleFav={[Function]}
      upVotes={30}
      views={120}
    >
      <Animation
        appear={true}
        name="fade"
        timeOut={600}
      >
        <CSSTransitionGroup
          transitionAppear={true}
          transitionAppearTimeout={600}
          transitionEnter={true}
          transitionEnterTimeout={600}
          transitionLeave={true}
          transitionLeaveTimeout={600}
          transitionName="fade"
        >
          <TransitionGroup
            childFactory={[Function]}
            component="span"
            transitionAppear={true}
            transitionAppearTimeout={600}
            transitionEnter={true}
            transitionEnterTimeout={600}
            transitionLeave={true}
            transitionLeaveTimeout={600}
            transitionName="fade"
          >
            <span>
              <CSSTransitionGroupChild
                appear={true}
                appearTimeout={600}
                enter={true}
                enterTimeout={600}
                key=".0"
                leave={true}
                leaveTimeout={600}
                name="fade"
              >
                <div
                  className="recipe-card"
                  key=".0"
                >
                  <div
                    className="recipe-card-body"
                  >
                    <Image
                      className="recipe-card-img"
                      cloudName="resycom"
                      crop="fill"
                      height="250"
                      publicId="recipe-card-placeholder_ta9ikp_iebsf2.jpg"
                      width="400"
                    >
                      <img
                        className="recipe-card-img"
                        height="250"
                        src="http://res.cloudinary.com/resycom/image/upload/c_fill,h_250,w_400/recipe-card-placeholder_ta9ikp_iebsf2.jpg"
                        width="400"
                      />
                    </Image>
                    <div
                      className="recipe-card-underlay"
                    >
                      <p
                        className="lead text-capitalize"
                      >
                        dinner
                      </p>
                      <Button
                        Tag="button"
                        className="recipe-card-underlay-btn"
                        dataTarget=""
                        dataToggle=""
                        handleClick={[Function]}
                        href=""
                        id=""
                        text="view details"
                        type="button"
                      >
                        <button
                          className="btn recipe-card-underlay-btn"
                          data-dismiss="modal"
                          data-target=""
                          data-toggle=""
                          href=""
                          id=""
                          onClick={[Function]}
                          type="button"
                        >
                          view details
                        </button>
                      </Button>
                    </div>
                  </div>
                  <div
                    className="recipe-card-footer"
                  >
                    <h4
                      className="ellipsis"
                    >
                      rice and beans
                    </h4>
                    <div
                      className="d-flex col-12 no-padding justify-content-between"
                    >
                      <div
                        className="d-flex col-8 no-padding"
                      >
                        <Icon
                          active=""
                          className=""
                          dataTarget=""
                          dataToggle=""
                          handleClick={[Function]}
                          iconClass="fa fa-thumbs-up recipe-card-icon "
                          onMouseEnter={[Function]}
                          parentClass=""
                          pointer={false}
                          tip="Total Upvotes: 30"
                          toolTip={true}
                          toolTipId="1upvote"
                          toolTipPlace="bottom"
                          toolTipType="dark"
                        >
                          <div
                            className="icon-wrapper"
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                            role="button"
                            tabIndex="0"
                          >
                            <i
                              className="icon fa fa-thumbs-up recipe-card-icon "
                              data-for="1upvote"
                              data-target=""
                              data-tip="Total Upvotes: 30"
                              data-toggle=""
                            >
                              <span>
                                30
                              </span>
                            </i>
                            <ReactTooltip
                              effect="solid"
                              id="1upvote"
                              insecure={true}
                              place="bottom"
                              resizeHide={true}
                              type="dark"
                              wrapper="div"
                            >
                              <div
                                className="__react_component_tooltip place-top type-dark "
                                data-id="tooltip"
                              />
                            </ReactTooltip>
                          </div>
                        </Icon>
                        <Icon
                          active=""
                          className=""
                          dataTarget=""
                          dataToggle=""
                          handleClick={[Function]}
                          iconClass="fa fa-thumbs-down recipe-card-icon "
                          onMouseEnter={[Function]}
                          parentClass=""
                          pointer={false}
                          tip="Total Downvotes: 0"
                          toolTip={true}
                          toolTipId="1downvote"
                          toolTipPlace="bottom"
                          toolTipType="dark"
                        >
                          <div
                            className="icon-wrapper"
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                            role="button"
                            tabIndex="0"
                          >
                            <i
                              className="icon fa fa-thumbs-down recipe-card-icon "
                              data-for="1downvote"
                              data-target=""
                              data-tip="Total Downvotes: 0"
                              data-toggle=""
                            >
                              <span>
                                0
                              </span>
                            </i>
                            <ReactTooltip
                              effect="solid"
                              id="1downvote"
                              insecure={true}
                              place="bottom"
                              resizeHide={true}
                              type="dark"
                              wrapper="div"
                            >
                              <div
                                className="__react_component_tooltip place-top type-dark "
                                data-id="tooltip"
                              />
                            </ReactTooltip>
                          </div>
                        </Icon>
                        <Icon
                          active=""
                          className=""
                          dataTarget=""
                          dataToggle=""
                          handleClick={[Function]}
                          iconClass="fa fa-eye recipe-card-icon"
                          onMouseEnter={[Function]}
                          parentClass=""
                          pointer={false}
                          tip="Total Number of Views: 120"
                          toolTip={true}
                          toolTipId="1views"
                          toolTipPlace="bottom"
                          toolTipType="dark"
                        >
                          <div
                            className="icon-wrapper"
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                            role="button"
                            tabIndex="0"
                          >
                            <i
                              className="icon fa fa-eye recipe-card-icon"
                              data-for="1views"
                              data-target=""
                              data-tip="Total Number of Views: 120"
                              data-toggle=""
                            >
                              <span>
                                120
                              </span>
                            </i>
                            <ReactTooltip
                              effect="solid"
                              id="1views"
                              insecure={true}
                              place="bottom"
                              resizeHide={true}
                              type="dark"
                              wrapper="div"
                            >
                              <div
                                className="__react_component_tooltip place-top type-dark "
                                data-id="tooltip"
                              />
                            </ReactTooltip>
                          </div>
                        </Icon>
                      </div>
                      <Icon
                        active=""
                        className=""
                        dataTarget=""
                        dataToggle=""
                        handleClick={[Function]}
                        iconClass="fa recipe-card-icon fav-color fav fa-heart-o"
                        onMouseEnter={[Function]}
                        parentClass=""
                        pointer={true}
                        tip="Favourite This Recipe"
                        toolTip={true}
                        toolTipId="1fav"
                        toolTipPlace="bottom"
                        toolTipType="dark"
                      >
                        <div
                          className="icon-wrapper pointer"
                          onClick={[Function]}
                          onMouseEnter={[Function]}
                          role="button"
                          tabIndex="0"
                        >
                          <i
                            className="icon fa recipe-card-icon fav-color fav fa-heart-o"
                            data-for="1fav"
                            data-target=""
                            data-tip="Favourite This Recipe"
                            data-toggle=""
                          >
                            <span />
                          </i>
                          <ReactTooltip
                            effect="solid"
                            id="1fav"
                            insecure={true}
                            place="bottom"
                            resizeHide={true}
                            type="dark"
                            wrapper="div"
                          >
                            <div
                              className="__react_component_tooltip place-top type-dark "
                              data-id="tooltip"
                            />
                          </ReactTooltip>
                        </div>
                      </Icon>
                    </div>
                  </div>
                </div>
              </CSSTransitionGroupChild>
            </span>
          </TransitionGroup>
        </CSSTransitionGroup>
      </Animation>
    </RecipeCard>
  </Router>
</BrowserRouter>
`;

exports[`Recipes card component  render as expected when favorite icon is clicked  2`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <RecipeCard
      category="dinner"
      downVotes={0}
      image={Object {}}
      isFav={[Function]}
      push={[Function]}
      recipeId={1}
      recipeName="rice and beans"
      toggleFav={[Function]}
      upVotes={30}
      views={120}
    >
      <Animation
        appear={true}
        name="fade"
        timeOut={600}
      >
        <CSSTransitionGroup
          transitionAppear={true}
          transitionAppearTimeout={600}
          transitionEnter={true}
          transitionEnterTimeout={600}
          transitionLeave={true}
          transitionLeaveTimeout={600}
          transitionName="fade"
        >
          <TransitionGroup
            childFactory={[Function]}
            component="span"
            transitionAppear={true}
            transitionAppearTimeout={600}
            transitionEnter={true}
            transitionEnterTimeout={600}
            transitionLeave={true}
            transitionLeaveTimeout={600}
            transitionName="fade"
          >
            <span>
              <CSSTransitionGroupChild
                appear={true}
                appearTimeout={600}
                enter={true}
                enterTimeout={600}
                key=".0"
                leave={true}
                leaveTimeout={600}
                name="fade"
              >
                <div
                  className="recipe-card"
                  key=".0"
                >
                  <div
                    className="recipe-card-body"
                  >
                    <Image
                      className="recipe-card-img"
                      cloudName="resycom"
                      crop="fill"
                      height="250"
                      publicId="recipe-card-placeholder_ta9ikp_iebsf2.jpg"
                      width="400"
                    >
                      <img
                        className="recipe-card-img"
                        height="250"
                        src="http://res.cloudinary.com/resycom/image/upload/c_fill,h_250,w_400/recipe-card-placeholder_ta9ikp_iebsf2.jpg"
                        width="400"
                      />
                    </Image>
                    <div
                      className="recipe-card-underlay"
                    >
                      <p
                        className="lead text-capitalize"
                      >
                        dinner
                      </p>
                      <Button
                        Tag="button"
                        className="recipe-card-underlay-btn"
                        dataTarget=""
                        dataToggle=""
                        handleClick={[Function]}
                        href=""
                        id=""
                        text="view details"
                        type="button"
                      >
                        <button
                          className="btn recipe-card-underlay-btn"
                          data-dismiss="modal"
                          data-target=""
                          data-toggle=""
                          href=""
                          id=""
                          onClick={[Function]}
                          type="button"
                        >
                          view details
                        </button>
                      </Button>
                    </div>
                  </div>
                  <div
                    className="recipe-card-footer"
                  >
                    <h4
                      className="ellipsis"
                    >
                      rice and beans
                    </h4>
                    <div
                      className="d-flex col-12 no-padding justify-content-between"
                    >
                      <div
                        className="d-flex col-8 no-padding"
                      >
                        <Icon
                          active=""
                          className=""
                          dataTarget=""
                          dataToggle=""
                          handleClick={[Function]}
                          iconClass="fa fa-thumbs-up recipe-card-icon "
                          onMouseEnter={[Function]}
                          parentClass=""
                          pointer={false}
                          tip="Total Upvotes: 30"
                          toolTip={true}
                          toolTipId="1upvote"
                          toolTipPlace="bottom"
                          toolTipType="dark"
                        >
                          <div
                            className="icon-wrapper"
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                            role="button"
                            tabIndex="0"
                          >
                            <i
                              className="icon fa fa-thumbs-up recipe-card-icon "
                              data-for="1upvote"
                              data-target=""
                              data-tip="Total Upvotes: 30"
                              data-toggle=""
                            >
                              <span>
                                30
                              </span>
                            </i>
                            <ReactTooltip
                              effect="solid"
                              id="1upvote"
                              insecure={true}
                              place="bottom"
                              resizeHide={true}
                              type="dark"
                              wrapper="div"
                            >
                              <div
                                className="__react_component_tooltip place-top type-dark "
                                data-id="tooltip"
                              />
                            </ReactTooltip>
                          </div>
                        </Icon>
                        <Icon
                          active=""
                          className=""
                          dataTarget=""
                          dataToggle=""
                          handleClick={[Function]}
                          iconClass="fa fa-thumbs-down recipe-card-icon "
                          onMouseEnter={[Function]}
                          parentClass=""
                          pointer={false}
                          tip="Total Downvotes: 0"
                          toolTip={true}
                          toolTipId="1downvote"
                          toolTipPlace="bottom"
                          toolTipType="dark"
                        >
                          <div
                            className="icon-wrapper"
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                            role="button"
                            tabIndex="0"
                          >
                            <i
                              className="icon fa fa-thumbs-down recipe-card-icon "
                              data-for="1downvote"
                              data-target=""
                              data-tip="Total Downvotes: 0"
                              data-toggle=""
                            >
                              <span>
                                0
                              </span>
                            </i>
                            <ReactTooltip
                              effect="solid"
                              id="1downvote"
                              insecure={true}
                              place="bottom"
                              resizeHide={true}
                              type="dark"
                              wrapper="div"
                            >
                              <div
                                className="__react_component_tooltip place-top type-dark "
                                data-id="tooltip"
                              />
                            </ReactTooltip>
                          </div>
                        </Icon>
                        <Icon
                          active=""
                          className=""
                          dataTarget=""
                          dataToggle=""
                          handleClick={[Function]}
                          iconClass="fa fa-eye recipe-card-icon"
                          onMouseEnter={[Function]}
                          parentClass=""
                          pointer={false}
                          tip="Total Number of Views: 120"
                          toolTip={true}
                          toolTipId="1views"
                          toolTipPlace="bottom"
                          toolTipType="dark"
                        >
                          <div
                            className="icon-wrapper"
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                            role="button"
                            tabIndex="0"
                          >
                            <i
                              className="icon fa fa-eye recipe-card-icon"
                              data-for="1views"
                              data-target=""
                              data-tip="Total Number of Views: 120"
                              data-toggle=""
                            >
                              <span>
                                120
                              </span>
                            </i>
                            <ReactTooltip
                              effect="solid"
                              id="1views"
                              insecure={true}
                              place="bottom"
                              resizeHide={true}
                              type="dark"
                              wrapper="div"
                            >
                              <div
                                className="__react_component_tooltip place-top type-dark "
                                data-id="tooltip"
                              />
                            </ReactTooltip>
                          </div>
                        </Icon>
                      </div>
                      <Icon
                        active=""
                        className=""
                        dataTarget=""
                        dataToggle=""
                        handleClick={[Function]}
                        iconClass="fa recipe-card-icon fav-color fav fa-heart-o"
                        onMouseEnter={[Function]}
                        parentClass=""
                        pointer={true}
                        tip="Favourite This Recipe"
                        toolTip={true}
                        toolTipId="1fav"
                        toolTipPlace="bottom"
                        toolTipType="dark"
                      >
                        <div
                          className="icon-wrapper pointer"
                          onClick={[Function]}
                          onMouseEnter={[Function]}
                          role="button"
                          tabIndex="0"
                        >
                          <i
                            className="icon fa recipe-card-icon fav-color fav fa-heart-o"
                            data-for="1fav"
                            data-target=""
                            data-tip="Favourite This Recipe"
                            data-toggle=""
                          >
                            <span />
                          </i>
                          <ReactTooltip
                            effect="solid"
                            id="1fav"
                            insecure={true}
                            place="bottom"
                            resizeHide={true}
                            type="dark"
                            wrapper="div"
                          >
                            <div
                              className="__react_component_tooltip place-top type-dark "
                              data-id="tooltip"
                            />
                          </ReactTooltip>
                        </div>
                      </Icon>
                    </div>
                  </div>
                </div>
              </CSSTransitionGroupChild>
            </span>
          </TransitionGroup>
        </CSSTransitionGroup>
      </Animation>
    </RecipeCard>
  </Router>
</BrowserRouter>
`;
